src/pool.spec.ts

```diff
@@ -401,6 +401,62 @@ describe("Constant product pool", () => {
     await pool.updateState();
     expect(pool.state.totalLiquidity).toBe(1000);
   });
+
+  it("zap e2e scenario", async () => {
+    const { account, pool } = await makeFreshTestBed({
+      poolType: "CONSTANT_PRODUCT",
+    });
+
+    expect(pool.state).toEqual({
+      totalLiquidity: 0,
+      totalPrimary: 0,
+      totalSecondary: 0,
+      primaryAssetPrice: 0,
+      secondaryAssetPrice: 0,
+    });
+
+    // Opt in for liquidity asset
+    const liqOptInTx = await pool.liquidityAsset.prepareOptInTx(account.addr);
+    await signAndSend(liqOptInTx, account);
+
+    // Add first liquidity
+    const liquidityAddition = pool.prepareAddLiquidity({
+      primaryAssetAmount: 100_000,
+      secondaryAssetAmount: 100_000,
+    });
+    const addLiqTxGroup = await liquidityAddition.prepareTxGroup(account.addr);
+    expect(addLiqTxGroup.transactions.length).toBe(3);
+    await signAndSend(addLiqTxGroup, account);
+    await pool.updateState();
+    expect(pool.state).toEqual({
+      totalLiquidity: 100_000,
+      totalPrimary: 100_000,
+      totalSecondary: 100_000,
+      primaryAssetPrice: 1,
+      secondaryAssetPrice: 1,
+    });
+
+    // Zapping
+    const zapParams = pool.getZapParams({ secondaryAssetAmount: 10_000 });
+    expect(zapParams.swapDeposited + zapParams.secondaryAddLiq).toBe(10_000n);
+    expect(zapParams.swapReceived >= zapParams.primaryAddLiq).toBe(true);
+
+    const zapTxGroup = await pool.prepareZapTxGroup({
```

Comment:
IMO, the zaps should be modeled similarly to swaps and adding liquidity.

How about:
```ts
const zap = pool.prepareZap({secondaryAssetAmount: 10_000});

// Zap is combination of a swap and adding liquidity so
zap.swap // a Swap object
zap.liquidityAddition // a LiquidityAddition object

// The following is valid.
const txs = [
  ...pool.buildSwapTxs({swap: zap.swap, ...}),
  ...pool.buildAddLiquidityTxs({liquidityAddition: zap.liquidityAddition, ...}),
]

// There's a convenience method for executing a zap.
const group = zap.prepareTxGroup(...)
```

The zap logic should reside in a separate file `zap.ts` similarly to swaps and adding liquidity.